### PR TITLE
Recode verification system

### PR DESCRIPTION
**Pull request by jonesdevelopment**
- Restructures the verification session handling
- Makes the code less complex/more readable

### CODE REVIEW

**Comment by jonesdevelopment** [comment]
There are some "issues" with 1.18.2-1.19.3 clients where it doesn't instantly render the world... It's stuck in "Downloading world" (or something similar) for some time.

File: sonar-common/src/main/java/xyz/jonesdev/sonar/common/fallback/session/FallbackGravitySessionHandler.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    // Make sure the client knows that we're sending chunks next
+    if (user.getProtocolVersion().compareTo(MINECRAFT_1_20_3) >= 0) {
+      user.delayedWrite(START_WRITING_CHUNKS);
+    }
+    // Teleport player into an empty world by sending an empty chunk packet
+    user.delayedWrite(EMPTY_CHUNK_DATA);
+    // Spawn the invisible platform below the player
+    if (this.enableCollisionsCheck) {
+      user.delayedWrite(updateSectionBlocks);
+    }
```
Comment (jonesdevelopment) [inline sonar-common/src/main/java/xyz/jonesdev/sonar/common/fallback/session/FallbackGravitySessionHandler.java:70]: Related to the 1.18.2-1.19.3 bug

**Comment by jonesdevelopment** [comment]
@FallenCrystal Could you please test this by using Geyser and checking if it works as intended?

**Comment by FallenCrystal** [comment]
> @FallenCrystal Could you please test this by using Geyser and checking if it works as intended?

Failed to verification. `invalid gravity: 1, 263.9166900, 0.0768400000, 0.1552400000 != 0.0784000000` (sometimes it works. sometimes it doesn't. just like the problem we had before probably)

**Comment by jonesdevelopment** [comment]
> Failed to verification. `invalid gravity: 1, 263.9166900, 0.0768400000, 0.1552400000 != 0.0784000000` (sometimes it works. sometimes it doesn't. just like the problem we had before probably)

Alright. Thanks for letting me know. Seems like it doesn't exempt geyser players correctly.

**Comment by jonesdevelopment** [comment]
@FallenCrystal Can you please check if the issue still persists now?

**Comment by Dynamic-Gravity** [comment]
@jonesdevelopment  may I have either the compiled jar file or specific instructions to implement the velocity.json so I can test it out on my server to verify? See my disco thread for details, thank you.

**Comment by FallenCrystal** [comment]
> @FallenCrystal Can you please check if the issue still persists now?

kicked by `didn't send plugin message`

~~I'll change some code on my branch and push the changes after that.~~

**Comment by jonesdevelopment** [comment]
> kicked by didn't send plugin message

Should be fixed now.

**Comment by jonesdevelopment** [comment]
whoops, sorry!

**Comment by FallenCrystal** [comment]
Looks pretty much the same as before. There shouldn't be any more problems.

btw It wouldn't be better if we stated that the plugin message was sent by the Geyser and not the client. (probably? It seems to only be sent after the player joins the game.)

File: sonar-common/src/main/java/xyz/jonesdev/sonar/common/fallback/session/FallbackGravitySessionHandler.java (hunk 1 above)
Comment (jonesdevelopment) [reply sonar-common/src/main/java/xyz/jonesdev/sonar/common/fallback/session/FallbackGravitySessionHandler.java:70]: I'll take a look at this later when I have some time for it. 1.18.2-1.19.3 can verify without problems, this is just a *potential* issue.

**Pull request merged**